tests/packet_tests.cpp

```diff
@@ -359,4 +359,311 @@ BOOST_AUTO_TEST_CASE(PacketValueConstructorRequest)
+
+    RadProto::Packet other(d.data(), d.size(), "secret");
+    RadProto::Packet p(other);
+
+    BOOST_CHECK_EQUAL(p.type(), 2);
+
+    BOOST_CHECK_EQUAL(p.id(), 0xd0);
+
+    std::array<uint8_t, 16> authExpected {0x93, 0xa9, 0x61, 0x8b, 0x2f, 0x4c, 0x5a, 0x51, 0x65, 0x67, 0x3d, 0xb4, 0x07, 0x30, 0xa2, 0x39};
+
+    BOOST_TEST(p.auth() == authExpected, boost::test_tools::per_element());
+
+    const auto& attrs = p.attributes();
+
+    auto* attr0 = findAttribute(attrs, RadProto::USER_NAME);
+
+    BOOST_REQUIRE(attr0 != nullptr);
+    BOOST_CHECK_EQUAL(attr0->type(), RadProto::USER_NAME);
+    BOOST_CHECK_EQUAL(attr0->toString(), "test");
+
+    auto* attr1 = findAttribute(attrs, RadProto::NAS_PORT);
+
+    BOOST_REQUIRE(attr1 != nullptr);
+    BOOST_CHECK_EQUAL(attr1->type(), RadProto::NAS_PORT);
+    BOOST_CHECK_EQUAL(attr1->toString(), "20");
+
+    auto* attr2 = findAttribute(attrs, RadProto::NAS_IP_ADDRESS);
+
+    BOOST_REQUIRE(attr2 != nullptr);
+    BOOST_CHECK_EQUAL(attr2->type(), RadProto::NAS_IP_ADDRESS);
+    BOOST_CHECK_EQUAL(attr2->toString(), "127.104.22.17");
+
+    auto* attr3 = findAttribute(attrs, RadProto::CALLBACK_NUMBER);
+
+    BOOST_REQUIRE(attr3 != nullptr);
+    BOOST_CHECK_EQUAL(attr3->type(), RadProto::CALLBACK_NUMBER);
+    BOOST_CHECK_EQUAL(attr3->toString(), "313233616263");
+
+    auto* attr4 = findAttribute(attrs, RadProto::CHAP_PASSWORD);
+
+    BOOST_REQUIRE(attr4 != nullptr);
+    BOOST_CHECK_EQUAL(attr4->type(), RadProto::CHAP_PASSWORD);
+    BOOST_CHECK_EQUAL(attr4->toString(), "1 31323334353637383961626364656667");
+
+    std::vector<RadProto::VendorSpecific> vendor = p.vendorSpecific();
+
+    BOOST_REQUIRE_EQUAL(vendor.size(), 1);
+    BOOST_CHECK_EQUAL(vendor[0].vendorId(), 171);
+    BOOST_CHECK_EQUAL(vendor[0].vendorType(), 1);
+    BOOST_CHECK_EQUAL(vendor[0].toString(), "00000003");
+
+    std::set<uint8_t> types {RadProto::USER_NAME, RadProto::NAS_PORT, RadProto::NAS_IP_ADDRESS, RadProto::CALLBACK_NUMBER, RadProto::CHAP_PASSWORD};
+
+    BOOST_REQUIRE_EQUAL(attrs.size(), 5);
+    BOOST_CHECK(types.count(attrs[0]->type()) == 1);
+    BOOST_CHECK(types.count(attrs[1]->type()) == 1);
+    BOOST_CHECK(types.count(attrs[2]->type()) == 1);
+    BOOST_CHECK(types.count(attrs[3]->type()) == 1);
+    BOOST_CHECK(types.count(attrs[4]->type()) == 1);
+
+    std::vector<uint8_t> values = p.makeSendBuffer("secret");
+
+    BOOST_REQUIRE_EQUAL(values.size(), 77);
+
+    BOOST_REQUIRE_EQUAL(d.size(), 77);
+
+    BOOST_TEST(values == d, boost::test_tools::per_element());
+}
+
+BOOST_AUTO_TEST_CASE(PacketValueCopyConstructorResponse)
```

Comment:
There is no difference in how the copy constructor works, whether the `other` is constructed from a buffer, or piecewise. You do not need to check it.